### ISSUE TITLE
signing key change

### ISSUE THREAD

**Issue by IzzySoft** [DanielRendox/GroceryGenius#18]
I see the signing key changed – and thanks for you transparently mention that with the release notes! However, to ensure it's really you it would be good to have some proof (see e.g. [How to keep your key safe and what measures to take for the event of loss?](https://f-droid.org/2023/09/03/reproducible-builds-signing-keys-and-binary-repos.html#lessons-learned-2-how-to-keep-your-key-safe-and-what-measures-to-take-for-the-event-of-loss) for some background – and again, this is no accusation; I also saw you already took measures to prevent similar things happen again by the means of backups).

While your commits are all signed, they are just signed by Github's key – so anyone who might have "hacked into your account" could do that. I guess there's also nobody who could vouch for you, as you're the sole contributor here and no alternative contact has been established.

However, you have a second app here: Routinetracker, which uses its own key. Hopefully that was not lost as well? If you still have it: could you please sign the v0.1.2 APK with that key and attach it here to this issue? That could then be used as verification: the two APKs then should only differ in the signing. We've got tools to verify that. As added plus you could then refer here for proof :wink: 

----

As you might wonder where I come from and why I'm asking for this: your app is [listed at the IzzyOnDroid repo](https://apt.izzysoft.de/packages/com.rendox.routinetracker), where the signing keys are pinned – so today's update triggered an alert (and the update was refused). We can fix that as soon as some proof has been established. Oh, and you can [pick a badge](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) to link there e.g. from your Readme then :wink: 

Thanks in advance for your support!

**Comment by DanielRendox** [comment]
Hi, @IzzySoft. Thank you for your support! Fortunately, the signing key for Routine Tracker has not been lost. So I was able to sign the v0.1.2 APK with it. Please check the attached files and let me know if it solves the issue.

[grocery-genius-v0.1.2-signing-check.zip](https://github.com/user-attachments/files/16900012/grocery-genius-v0.1.2-signing-check.zip) 
(I changed the file's extension to ZIP because GitHub does not allow uploading APKs here)

And here is the Routine Tracker's APK: [routinetracker-v0.1.0.apk](https://github.com/DanielRendox/RoutineTracker/releases/download/v0.1.0/routinetracker-v0.1.0.apk)

**Comment by IzzySoft** [comment]
Close enough, thanks!

```diff
  -rw-r--r--  0.0 unx       56 b-       52 defN 1981-01-01 01:01:02 8f478391 META-INF/com/android/build/gradle/app-metadata.properties
- -rw-r--r--  0.0 unx      120 b-      117 defN 1981-01-01 01:01:02 d9ed9bb7 META-INF/version-control-info.textproto
+ -rw-r--r--  0.0 unx      120 b-      118 defN 1981-01-01 01:01:02 808235b0 META-INF/version-control-info.textproto
  -rw-r--r--  0.0 unx    12145 b-    12145 stor 1981-01-01 01:01:02 9265ab9f assets/dexopt/baseline.prof
  -rw-r--r--  0.0 unx      867 b-      867 stor 1981-01-01 01:01:02 02ec2fff assets/dexopt/baseline.profm
  -rw-r--r--  0.0 unx  2042524 b-   994879 defN 1981-01-01 01:01:02 c2ed0925 classes.dex
@@ -825,7 +825,7 @@
  -rw----     0.0 fat      448 b-      219 defN 1981-01-01 01:01:02 0c0249f5 res/zp.xml
  -rw----     0.0 fat    61404 b-    61404 stor 1981-01-01 01:01:02 30e3e5dc res/zq.png
  -rw----     0.0 fat   855540 b-   855540 stor 1981-01-01 01:01:02 89977807 resources.arsc
- -rw-r--r--  0.0 unx    79467 b-    35076 defN 1981-01-01 01:01:02 7922152c META-INF/CERT.SF
- -rw-r--r--  0.0 unx     1347 b-     1078 defN 1981-01-01 01:01:02 0d9c83d9 META-INF/CERT.RSA
- -rw-r--r--  0.0 unx    79393 b-    32018 defN 1981-01-01 01:01:02 64cd23ce META-INF/MANIFEST.MF
+ -rw-r--r--  0.0 unx    79467 b-    35073 defN 1981-01-01 01:01:02 e08c5893 META-INF/CERT.SF
+ -rw-r--r--  0.0 unx     1264 b-     1050 defN 1981-01-01 01:01:02 d98ed7e2 META-INF/CERT.RSA
+ -rw-r--r--  0.0 unx    79393 b-    32014 defN 1981-01-01 01:01:02 4c84dd0b META-INF/MANIFEST.MF
```

Certs of course differ – and then you compiled from a different commit (one is from before, one after merging). Key matches the one from routinetracker (which I took from the copy in the repo here t IoD) – so I'd call that a very good and sound proof.

Will initiate the keyswitch then. For a few weeks, both keys will be valid. Then I'll remove the old APK and disable the old key. I've also added a releasenote saying folks have to uninstall and re-install due to key change.

![image](https://github.com/user-attachments/assets/1af4db8b-e463-4162-8130-555c261377c3)

While I'm here:

![image](https://github.com/user-attachments/assets/6aab207a-3fd2-4d98-9138-56471aac2b33)

This can easily be avoided with a minor addition to your `build.gradle`:

```gradle
android {
    dependenciesInfo {
        // Disables dependency metadata when building APKs.
        includeInApk = false
        // Disables dependency metadata when building Android App Bundles.
        includeInBundle = false
    }
}
```

For some background: that BLOB is supposed to be just a binary representation of your app's dependency tree. But as it's encrypted with a public key belonging to Google, only Google can read it – and nobody else can even verify what it really contains. More details can be found e.g. here: [Ramping up security: additional APK checks are in place with the IzzyOnDroid repo](https://android.izzysoft.de/articles/named/iod-scan-apkchecks#blobs).

And one last thing: please keep the file naming pattern for the APK files at your releases constant. Else the verification builder here might not find them, and your nice green shield for [Reproducible Builds](https://reproducible-builds.org/) is lost (see also [Reproducible Builds, special client support and more at IzzyOnDroid](https://android.izzysoft.de/articles/named/iod-rbs-mirrors-clients)) :wink: I just tweaked the name to adjust for the missing `-`, so it now accepts both variants. But please no third variant :see_no_evil: Ugh, and that APK failed RB – it was not built from the commit the release tag points to (embeded commit hash points to b09064ae636347ae34c4fd3a6557770b6f1bce16). "First basic rule" :wink: Unfortunately it fails from that commit as well – dirty tree maybe?

```diff
  -rw-r--r--  0.0 unx      120 b-      117 defN 1981-01-01 01:01:02 d9ed9bb7 META-INF/version-control-info.textproto
- -rw-r--r--  0.0 unx    12145 b-    12145 stor 1981-01-01 01:01:02 9265ab9f assets/dexopt/baseline.prof
- -rw-r--r--  0.0 unx      867 b-      867 stor 1981-01-01 01:01:02 02ec2fff assets/dexopt/baseline.profm
- -rw-r--r--  0.0 unx  2042524 b-   994879 defN 1981-01-01 01:01:02 c2ed0925 classes.dex
- -rw-r--r--  0.0 unx  3357020 b-  1607148 defN 1981-01-01 01:01:02 c5755260 classes2.dex
+ -rw-r--r--  0.0 unx    12170 b-    12170 stor 1981-01-01 01:01:02 647709da assets/dexopt/baseline.prof
+ -rw-r--r--  0.0 unx      865 b-      865 stor 1981-01-01 01:01:02 055e48d9 assets/dexopt/baseline.profm
+ -rw-r--r--  0.0 unx  2042300 b-   994736 defN 1981-01-01 01:01:02 452842d7 classes.dex
+ -rw-r--r--  0.0 unx  3355824 b-  1607059 defN 1981-01-01 01:01:02 74da9849 classes2.dex
  -rw-r--r--  0.0 unx     7112 b-     2630 defN 1981-01-01 01:01:02 4f56ac38 lib/arm64-v8a/libdatastore_shared_counter.so
```

Diff of `classes.dex` is rather huge. Afraid we have to leave this one on "RB failed" and hope the next release will be clean again.

PS: be welcome to [pick a badge](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) to give that lonely Github one a companion, linking it [to your app at IzzyOnDroid](https://apt.izzysoft.de/packages/com.rendox.grocerygenius) :smiley:

**Comment by DanielRendox** [comment]
Hey, @IzzySoft. I've just published a new release `v0.1.3`. Now the build should be reproducible.

**Comment by IzzySoft** [comment]
Confirmed, thanks!

Btw: you might wish to [pick a badge](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) to place next to the lonely Github one in your Readme and link to [your app at IzzyOnDroid](https://apt.izzysoft.de/packages/com.rendox.grocerygenius) :smiley: With the next sync (around 6 pm UTC) the new release should show up there and carry the green shield again.

Feel free to close the issue afterwards – I'd say we've got everything solved now, right? Thanks for your help with that!
